### PR TITLE
[Clustering] Add the layer support for StackedRNNCells, PeepholeLSTMCell, and Bidirectional.

### PR DESCRIPTION
**Pull request by Ruomei**
This PR adds support for StackedRNNCells, PeepholeLSTMCell, and Bidirectional in clustering. To verify the implementation, unit tests and end-to-end tests are provided in the PR.
Below are numbers generated using the imdb_multiple_cells.py. In those experiments, the clustering wrapper was applied to a model with randomly initialized weights for 1 epoch on the IMDB dataset with a number of clusters equal to 16.

| Types of RNN layers/Cells 	| Experimental setups                               	| Accuracy in clustering experiments 	|
|---------------------------	|---------------------------------------------------	|------------------------------------	|
| StackedRNNCell            	|Two LSTMCells wrapped by a StackedRNNCell | 0.8502                             	|
| PeepholeLSTMCell          	| Two PeepholeLSTMCells wrapped by a StackedRNNCell | 0.8403                             |
| Bidirectional             	| A LSTM wrapped by a Bidirectional  | 0.8492 |

### CODE REVIEW

**Comment by xhae** [comment]
@daverim can you take a look on this?

**Review by wwwind — APPROVED** [review]

File: tensorflow_model_optimization/python/core/clustering/keras/clustering_registry_test.py [hunk 1]
```diff
@@ -433,15 +500,16 @@ def testMakeClusterableDoesNotWorksOnKerasRNNLayerWithClusterableCell(self):
 
   def testMakeClusterableRaisesErrorOnRNNLayersUnsupportedCell(self):
     """Verifies that make_clusterable() raises an exception.
-
```
Comment (daverim) [inline tensorflow_model_optimization/python/core/clustering/keras/clustering_registry_test.py:436]: Nit: can you leave a space here?

File: tensorflow_model_optimization/python/examples/clustering/keras/imdb/BUILD [hunk 2]
```diff
@@ -50,3 +50,32 @@ py_binary(
...
+)
+
+py_binary(
+    name = "imdb_multiple_cells",
+    srcs = [
+        "imdb_multiple_cells.py",
+    ],
+    python_version = "PY3",
+    deps = [
+        # numpy dep1,
```
Comment (daverim) [inline tensorflow_model_optimization/python/examples/clustering/keras/imdb/BUILD:75]: what do these comments refer to?

File: tensorflow_model_optimization/python/core/clustering/keras/clustering_registry.py [hunk 3]
```diff
@@ -127,14 +162,33 @@ def get_clusterable_weights():
...
+        # when are wrapped by the StackedRNNCell layer
+        # The weight names will have indices attached only
+        # for the registry
+        if cell.__class__ in cls._SUPPORTED_RNN_CELLS:
+          cell_weights = []
+          cell_weights.append(('kernel/' + str(i), cell.kernel))
+          cell_weights.append(('recurrent_kernel/' + str(i),
+                               cell.recurrent_kernel))
+          return cell_weights
+
```
Comment (daverim) [inline tensorflow_model_optimization/python/core/clustering/keras/clustering_registry.py:176]: Nit: would this not be more clear with 
```
return [('kernel/' + str(i), cell.kernel),  ('recurrent_kernel/' + stri(i), cell.recurrent_kernel)]
``` 
As previously?

File: tensorflow_model_optimization/python/core/clustering/keras/cluster_wrapper.py [hunk 4]
```diff
@@ -321,10 +339,51 @@ def set_weights(self, weights):
...
+  """This wrapper augments a RNN layer so that the weights can be clustered.
+
+  The weight_name of a single cell in RNN layers is marked with an index in
+  registry. In the wrapper layer, the index needs to be removed for matching
+  the attribute of the cell layer."""
+  def get_weight_name_without_index(self, weight_name):
+    weight_name_no_index = weight_name.split('/')[0]
+    i = int(weight_name.split('/')[1])
+    return weight_name_no_index, i
+
```
Comment (daverim) [inline tensorflow_model_optimization/python/core/clustering/keras/cluster_wrapper.py:351]: Nit: would this not be better to be 
``` 
weight_name_with_index = weight_name.split('/')
return weight_name_with_index[0], int(weight_name_with_index[1])
```

**Review by daverim — COMMENTED** [review]
Looks good to me, a couple of really tiny nits.

File: tensorflow_model_optimization/python/core/clustering/keras/clustering_registry_test.py (hunk 1 above)
Comment (Ruomei) [reply tensorflow_model_optimization/python/core/clustering/keras/clustering_registry_test.py]: Yes, sure.

File: tensorflow_model_optimization/python/examples/clustering/keras/imdb/BUILD (hunk 2 above)
Comment (Ruomei) [reply tensorflow_model_optimization/python/examples/clustering/keras/imdb/BUILD]: They are not essential so I deleted them in all places. Thanks.

File: tensorflow_model_optimization/python/core/clustering/keras/clustering_registry.py (hunk 3 above)
Comment (Ruomei) [reply tensorflow_model_optimization/python/core/clustering/keras/clustering_registry.py:173]: Thanks, done.

File: tensorflow_model_optimization/python/core/clustering/keras/cluster_wrapper.py (hunk 4 above)
Comment (Ruomei) [reply tensorflow_model_optimization/python/core/clustering/keras/cluster_wrapper.py:350]: Thanks, done.

**Review by daverim — APPROVED** [review]
